airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/python_registry/context.py

```diff
@@ -0,0 +1,105 @@
+from typing import Optional, Type
+
+from pipelines.airbyte_ci.connectors.context import PipelineContext
+from pipelines.airbyte_ci.connectors.publish.context import PublishConnectorContext
+from pipelines.consts import DEFAULT_PYTHON_PACKAGE_REGISTRY_URL
+
+
+@dataclass
+class PythonPackageMetadata:
+    name: Optional[str]
+    version: Optional[str]
+
+
+class PythonRegistryPublishContext(PipelineContext):
+    def __init__(
+        self,
+        pypi_token: str,
+        package_path: str,
+        report_output_prefix: str,
+        is_local: bool,
+        git_branch: str,
+        git_revision: str,
+        ci_report_bucket: Optional[str] = None,
+        registry: str = DEFAULT_PYTHON_PACKAGE_REGISTRY_URL,
+        gha_workflow_run_url: Optional[str] = None,
+        dagger_logs_url: Optional[str] = None,
+        pipeline_start_timestamp: Optional[int] = None,
+        ci_context: Optional[str] = None,
+        ci_gcs_credentials: Optional[str] = None,
+        package_name: Optional[str] = None,
+        version: Optional[str] = None,
+    ) -> None:
+        self.pypi_token = pypi_token
+        self.registry = registry
+        self.package_path = package_path
+        self.package_metadata = PythonPackageMetadata(package_name, version)
+
+        pipeline_name = f"Publish PyPI {package_path}"
+
+        super().__init__(
+            pipeline_name=pipeline_name,
+            report_output_prefix=report_output_prefix,
+            ci_report_bucket=ci_report_bucket,
+            is_local=is_local,
+            git_branch=git_branch,
+            git_revision=git_revision,
+            gha_workflow_run_url=gha_workflow_run_url,
+            dagger_logs_url=dagger_logs_url,
+            pipeline_start_timestamp=pipeline_start_timestamp,
+            ci_context=ci_context,
+            ci_gcs_credentials=ci_gcs_credentials,
+        )
+
+    @classmethod
+    async def from_publish_connector_context(
+        cls: Type["PythonRegistryPublishContext"], connector_context: PublishConnectorContext
+    ) -> Optional["PythonRegistryPublishContext"]:
+        """
+        Create a PythonRegistryPublishContext from a ConnectorContext.
+
+        The metadata of the connector is read from the current workdir to capture changes that are not yet published.
+        If pypi is not enabled, this will return None.
+        """
+
+        current_metadata = connector_context.connector.metadata
+        connector_context.logger.info(f"Current metadata: {str(current_metadata)}")
+        if (
+            "remoteRegistries" not in current_metadata
+            or "pypi" not in current_metadata["remoteRegistries"]
+            or not current_metadata["remoteRegistries"]["pypi"]["enabled"]
+        ):
+            return None
+
+        version = current_metadata["dockerImageTag"]
+        if connector_context.pre_release:
+            # use current date as pre-release version
+            release_candidate_tag = datetime.now().strftime("%Y%m%d%H%M")
```

Comment:
@flash1293 can you add a comment here explaining why using the same tagging strategy as docker tags does not work? 🙏